### PR TITLE
test cast string to timestamp correctness

### PR DESCRIPTION
**Pull request by res-life**
Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
Already tested year from 0001 to 9999. 
Depends on https://github.com/NVIDIA/spark-rapids-jni/pull/1539, so convert to draft.

File: tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZoneSuite.scala [hunk 1]
```diff
@@ -59,6 +60,24 @@ class TimeZoneSuite extends SparkQueryCompareTestSuite with BeforeAndAfterAll {
...
 
+  /**
+   * create string column vector
+   */
+  def createColumnVector(strings: Array[String]): ColumnVector = {
+    val rows = strings.length
+    withResource(HostColumnVector.builder(DType.STRING, rows)) { builder =>
+      var idx = 0
+      while (idx < rows) {
+        // convert seconds to micro seconds
```
Comment (thirtiseven) [inline tests/src/test/scala/com/nvidia/spark/rapids/timezone/TimeZoneSuite.scala:71]: Seems a invalid comment.

And nit: Is foreach better here?
```
strings.foreach { s =>
  builder.append(s)
}
```

**Comment by jlowe** [comment]
@res-life please retarget this to 24.04, as the spark-rapids-jni dependency is targeting 24.04.

**Comment by GaryShen2008** [comment]
@res-life  I think we can close this PR since we low prioritize the feature of cast string to timestamp. And revisit this PR when we work on the feature again.

**Comment by res-life** [comment]
Close it. Will open if it is needed.

**Pull request closed without merging**